src/containers/share-modal/ShareModal.tsx

```diff
@@ -23,79 +31,101 @@ const IS_NATIVE_MOBILE = process.env.REACT_APP_NATIVE_MOBILE
 
 export const ShareModal = () => {
   const [isOpen, setIsOpen] = useModalState('Share')
+
   const { toast } = useContext(ToastContext)
   const dispatch = useDispatch()
   const record = useRecord()
   const track = useSelector(getTrack)
-  const artist = useSelector((state: CommonState) =>
+  const { content, source } = useSelector(getShareState)
+  const trackArtist = useSelector((state: CommonState) =>
     getUser(state, { id: track?.owner_id })
   )
   const account = useSelector(getAccountUser)
-  const source = useSelector(getSource)
+
+  const { isEnabled: isShareSoundToTikTokEnabled } = useFlag(
+    FeatureFlags.SHARE_SOUND_TO_TIKTOK
+  )
 
   const isOwner = Boolean(
-    account && artist && account.user_id === artist.user_id
+    account && trackArtist && account.user_id === trackArtist.user_id
   )
 
   const handleClose = useCallback(() => {
     setIsOpen(false)
   }, [setIsOpen])
 
   const handleShareToTwitter = useCallback(() => {
-    if (track && artist && source) {
-      const twitterText = `Check out ${track.title} by ${artist.handle} on @AudiusProject #Audius`
-      const trackLink = getCopyableLink(track.permalink)
-      openTwitterLink(trackLink, twitterText)
-      handleClose()
+    let twitterText = ''
+    let link = ''
+    if (!source || !content) return
+    if (content.type === 'track') {
+      const { track, artist } = content
+      twitterText = `Check out ${track.title} by ${artist.handle} on @AudiusProject #Audius`
```

Comment:
For text displayed to users, we want to put it in a messages object for future internationalization like so: https://github.com/AudiusProject/audius-client/blob/master/src/components/banner/FanburstBanner.js#L10